ci/build.bat

```diff
@@ -1,27 +1,33 @@
-:: These are native MinGW builds, but they use the toolchain inside
-:: MSYS2, this allows using all the dependencies and tools available
-:: in MSYS2, but we cannot build inside the MSYS2 shell.
 echo on
 if "%CONFIGURATION%" == "MINGW_32" (
   set ARCH=i686
   set BITS=32
-) else (
+) else if "%CONFIGURATION%" == "MINGW_64" (
   set ARCH=x86_64
   set BITS=64
-)
-if "%CONFIGURATION%" == "MINGW_64-gcov" (
+) else if "%CONFIGURATION%" == "MINGW_64-gcov" (
   set USE_GCOV="-DUSE_GCOV=ON"
+) else if "%CONFIGURATION%" == "MSVC_32" (
+  set CMAKE_GENERATOR="Visual Studio 15 2017"
+) else if "%CONFIGURATION%" == "MSVC_64" (
+  set CMAKE_GENERATOR="Visual Studio 15 2017 Win64"
 )
 
-:: We cannot have sh.exe in the PATH (MinGW)
-set PATH=%PATH:C:\Program Files\Git\usr\bin;=%
-set PATH=C:\msys64\mingw%BITS%\bin;C:\Windows\System32;C:\Windows;%PATH%
-:: The default cpack in the PATH is not CMake
-set PATH=C:\Program Files (x86)\CMake\bin\cpack.exe;%PATH%
-
-:: Build third-party dependencies
-C:\msys64\usr\bin\bash -lc "pacman --verbose --noconfirm -Su" || goto :error
-C:\msys64\usr\bin\bash -lc "pacman --verbose --noconfirm --needed -S mingw-w64-%ARCH%-cmake mingw-w64-%ARCH%-perl mingw-w64-%ARCH%-diffutils mingw-w64-%ARCH%-unibilium gperf" || goto :error
+if "%CONFIGURATION:~0,5%" == "MINGW" (
+  :: These are native MinGW builds, but they use the toolchain inside
+  :: MSYS2, this allows using all the dependencies and tools available
+  :: in MSYS2, but we cannot build inside the MSYS2 shell.
+  set CMAKE_GENERATOR="MinGW Makefiles"
+  set CMAKE_GENERATOR_ARGS=VERBOSE=1
+  :: Add MinGW to the PATH and remove the Git directory because it
+  :: has a conflicting sh.exe
+  set "PATH=C:\msys64\mingw%BITS%\bin;C:\Windows\System32;C:\Windows;%PATH:C:\Program Files\Git\usr\bin;=%"
+  :: Build third-party dependencies
+  C:\msys64\usr\bin\bash -lc "pacman --verbose --noconfirm -Su" || goto :error
+  C:\msys64\usr\bin\bash -lc "pacman --verbose --noconfirm --needed -S mingw-w64-%ARCH%-cmake mingw-w64-%ARCH%-perl mingw-w64-%ARCH%-diffutils mingw-w64-%ARCH%-unibilium gperf" || goto :error
+) else if "%CONFIGURATION:~0,4%" == "MSVC" (
+  set CMAKE_GENERATOR_ARGS=/verbosity:detailed
```

Comment:
`/verbosity:detailed` looks like a 10x increase in log size. If we aren't debugging the build, let's use `/verbosity:normal` (which is the default, but leave it here explicitly so it's easy to change when needed).